File: frontend/lib/orderStatus.ts
export const STATUS_STEPS = ["pending", "confirmed", "preparing", "out_for_delivery", "delivered"] as const;

export const STATUS_LABELS: Record<string, string> = {
  pending: "Pending",
  confirmed: "Confirmed",
  preparing: "Preparing",
  out_for_delivery: "Out for Delivery",
  delivered: "Delivered",
  cancelled: "Cancelled",
};

// Admin list/detail endpoints return the C# enum name (PascalCase, e.g. "OutForDelivery")
// rather than the customer-facing wire format (snake_case, e.g. "out_for_delivery").
const PASCAL_TO_LABEL: Record<string, string> = {
  Pending: "Pending",
  Confirmed: "Confirmed",
  Preparing: "Preparing",
  OutForDelivery: "Out for Delivery",
  Delivered: "Delivered",
  Cancelled: "Cancelled",
};

export function labelForStatus(status: string): string {
  return STATUS_LABELS[status] ?? PASCAL_TO_LABEL[status] ?? status;
}

File: backend/CandyStation.Core/Enums/OrderStatus.cs
namespace CandyStation.Core.Enums;

public enum OrderStatus
{
    Pending,
    Confirmed,
    Preparing,
    OutForDelivery,
    Delivered,
    Cancelled,
}

public enum DeliveryMethod
{
    Delivery,
    Pickup,
}

public enum ItemStatus
{
    Active,
    Inactive,
}

/// <summary>Staff role stored on AdminUser. Admin has full CRUD access; Delivery is
/// order-status-and-print only (no product/category/bundle/customer/settings edit rights).</summary>
public enum StaffRole
{
    Admin,
    Delivery,
}

/// <summary>What a HeroSlide links to — exactly one of the slide's Product/Bundle FKs is set to match.</summary>
public enum HeroItemType
{
    Product,
    Bundle,
}
